### DISCUSSION TITLE
「文章修正提案機能」に関するご意見募集！

### DISCUSSION THREAD

**Discussion by kimkim0814** [increments/qiita-discussions#670] [アナウンス]
Qiita では、文章修正提案機能のルールとして[textlint-rule-ja-no-successive-word](https://github.com/textlint-ja/textlint-rule-ja-no-successive-word)を利用しています。
このルールは、同じ単語（形態素解析した Token）が間違って連続している箇所を見つけるための textlint ルールです。

現在、実用上の利便性を考慮して、同じ単語が連続していても除外するオプションを追加しようと考えています。
例えば、「よくよく」などは意図的に連続して使われることもありますので、このような表現も除外するようなオプションになります。

そこで、みなさまから除外したい単語のご意見をお聞かせいただきたいです。
ご提案お待ちしています!

**Comment by tanaka-asami415** [comment]
X や https://github.com/increments/qiita-discussions/discussions/701 でいただいた意見を反映し、連続する単語での修正提案機能のルールを緩和しました🚀
https://qiita.com/release-notes/2024-07-09

連続する単語以外でも、「文章修正提案機能」についてお気づきの事がありましたら
こちらのDiscussionsにご意見お寄せください✨

**Comment by mochi-yu** [comment]
文章修正提案機能の以下のエラーについて、エラーメッセージが日本語だと分かりやすく良いのかなと思いましたが、いかがでしょうか。

内容としては、略語をカッコ内に書くべきというもので、「Single Page Application（SPA）」と書かれることを期待しているようでした。

仕様を把握できておらず、このエラーメッセージがQiitaの仕様によるものなのか、Linterの仕様によるものなのか分かりませんが、ご確認いただけますと幸いです。

<img width="639" alt="image" src="https://github.com/user-attachments/assets/9d1087d4-429f-45fe-81f1-bc3467254d5e" />

**Reply by tomoasleep** [reply]
@mochi-yu
ご意見ありがとうございます！こちら返信が遅くなってしまいすみません :bow:

Qiita の文章修正提案機能についてなのですが、こちらは内部的には [textlint](https://textlint.github.io/) というオープンソースのツールと、オープンソースのルールプラグインを複数組み合わせて利用しております。

それで Qiita で表示しているエラーメッセージは、これらのエラーメッセージをそのまま表示しているため、ルール間で文体に違いが生じてしまったり、英語と日本語のメッセージが混在する状態になってしまっています。

こうした状態への対策として、Qiita 側で独自のエラーメッセージを用意して、それに差し替えるなどの方法が考えられるのですが、その場合、元のメッセージとQiita 独自のメッセージの対応関係を多数管理する必要が出てきてしまいます。
そうした場合の工数も鑑みて、現状はエラーメッセージに手は加えず、そのまま表示する、という仕様にさせていただいております。ご理解いただけますと幸いです :pray:

ただ、今後も様々な反応を見ながら、文章修正提案機能をより便利にお使いいただけるように改善は検討していきたいと考えていますので、気になった点は今後ともご意見いただけますと非常に嬉しいです :smiley:
